### DISCUSSION TITLE
How to configure elgot python like that?

### DISCUSSION THREAD

**Discussion by Thysrael** [joaotavora/eglot#1331] [Q&A]
I am trying to migrate from vscode to emacs, I have a `settings.json` like that

```json
{
    "python.analysis.extraPaths": [
        "${workspaceFolder}/site_scons",
        "${workspaceFolder}/src/python",
    ],
}
```

I know I can configure `eglot-workspace-configuration`, but I don't know how exactly, would you please help me to understand it?

**Comment by joaotavora** [comment]
You have to figure out exactly what language server you are using and then read at least this section in the manual: configuration: https://joaotavora.github.io/eglot/#Project_002dspecific-configuration

It includes examples.  After you have tried some things can back here with specific questions about what you are doing.

**Reply by Thysrael** [reply]
I use pyright, I write a `.dir-locals.el` at gem5 root , It looks like that

```lisp
((nil
  . ((eglot-workspace-configuration
      . (:pyright (:analysis (:extraPaths "./src/python/"))))
     )))
```
and it looks like not working.
